### PR TITLE
prevent generating invoices when order's distributor can't generate invoices

### PR DESCRIPTION
**Pull request by abdellani**
#### What? Why?

- related to https://github.com/openfoodfoundation/openfoodnetwork/issues/11896

When I worked on this [PR](https://github.com/openfoodfoundation/openfoodnetwork/pull/11918), we accidentally discovered that it's possible to generate invoices while the order's distributor can't generate invoices.
Check the following comments for more details:
https://github.com/openfoodfoundation/openfoodnetwork/pull/11918#issuecomment-1971434683
https://github.com/openfoodfoundation/openfoodnetwork/pull/11918#issuecomment-1971496252

#### What should we test?

1. open enterprise configuration for a distributor D1
2. enable taxes
3. set an ABN
4. open another tab (let's call it TAB2)
5. open an order that belongs to D1
6. open invoices 
image
7. return  to the previous TAB1
8. remove the ABN and save
9. return to TAB2
10. click on ACTIONS/PRINT
11. you should get a notification saying that "D1 must have a valid ABN before invoices can be used."
image

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/controllers/spree/admin/invoices_controller.rb [hunk 1]
```diff
@@ -19,8 +19,14 @@ def show
...
         @order = Order.find_by(number: params[:order_id])
-        authorize! :invoice, @order
-        OrderInvoiceGenerator.new(@order).generate_or_update_latest_invoice
+        if @order.distributor.can_invoice?
+          authorize! :invoice, @order
+          OrderInvoiceGenerator.new(@order).generate_or_update_latest_invoice
+        else
+          flash[:error] = t(:must_have_valid_business_number,
+                            enterprise_name: @order.distributor.name)
+        end
```
Comment (mkllnk) [inline app/controllers/spree/admin/invoices_controller.rb:28]: My first thought was why we don't include that check in the Abilities class to have one source of truth. But that wouldn't provide us with a good error message. I guess that's one downside of CanCan.

**Review by mkllnk — APPROVED** [review]

**Review by dacook — APPROVED** [review]
👍 Great

**Comment by drummer83** [comment]
Hi @abdellani,
We have a conflict here now. Probably from merging the previous PR around this topic?
Could you please have a look?
Thanks!

**Review by dacook — APPROVED** [review]
Thanks

**Review by mkllnk — APPROVED** [review]

**Comment by RachL** [comment]
Sorry for taking this long to test this one, I took the time to test other invoices related stuff, will open some issues about them #neverendingstory

This PR is ready to go! The message looks nice in the new v3 style :) 

image

**Pull request merged**